### DISCUSSION TITLE
Show which users homeserver allowed user to join `restricted`/`knock_restricted` room

### DISCUSSION THREAD

**Discussion by Mikaela** [element-hq/element-meta#546] [Ideas]
#### What would you like to do?

When I am using `knock_restricted` or `restricted` join rules and someone joins there, Nheko says `USERNAME liittyi käyttäjän @MXID:EXAMPLE.NET palvelimen suomalla vahvistuksella.` (`USERNAME joined with confirmation granted from users @MXID:EXAMPLE.NET homeserver` (sorry, backtranslating is difficult and it works in Finnish).

#### Why would you like to do it?

I am curious about this kind of information and Nheko shows it directly instead of having to view event source from Element Web.

#### How would you like to achieve it?

I would like to see it in the join message directly.

### Have you considered any alternatives?

It could stay in `/devtools` or event source being a bit cumbersome to access in my opinion, especially on Elements of other platforms.

### Additional context

I guess this issue should be in Element-meta, but I already wrote it here. Please feel free to transfer if you consider that appropiate.

**Comment by TheArcaneBrony** [comment]
`<user> joined via authorisation from <mxid>'s server.` is the English version of that string in Nheko.

I think it's useful information to have, given it also can alert ones of unintended permissions, for those aware of how restricted joins work at a high level.
